docs/resources/users_permission.md

```diff
@@ -0,0 +1,89 @@
+---
+page_title: "keycloak_users_permissions Resource"
+---
+
+# keycloak_users_permissions
+
+Allows you to manage all users Scope Based Permissions.
+
+This is part of a preview keycloak feature. You need to enable this feature to be able to use this resource.
+More information about enabling the preview feature can be found here: https://www.keycloak.org/docs/latest/server_installation/#profiles
+
+When enabling Users Permissions, Keycloak does several things automatically:
+1. Enable Authorization on build-in realm-management client
+1. Create scopes "view", "manage", "map-roles", "manage-group-membership", "impersonate", "user-impersonated"
+1. Create a resource representing the users permissions
+1. Create all scope based permission for the scopes and openid client resource
```

Comment:
Is this list accurate as well?  I should have asked this earlier.

Reply:
On a new realm, the "Autorization" tab is not available on the `realm-management` client. Unlike confidential clients (`account`, `broker`), it's not possible to set "Authorization Enabled" to ON.

Go to the tab `Permissions` of Users page and set it to "Enabled". You should see users permissions Scopes.
Return to the `realm-management` client, the "Autorization" tab is available.
In the sub-tabs of "Autorization" : 
- `Authorization Scopes`: the 6 scopes of user permissions are created
- `Resources` : the resource "Users" is created
- `Permissions` : A permission are created by resource and scope combination

Reply:
This is really helpful, thanks for the explanation!  I'm starting to understand the scope of this PR much better now.

So it looks like this resource, in general, allows you to add a single policy to each of these permissions.

![image](https://user-images.githubusercontent.com/3793752/97723834-32882100-1a9a-11eb-91f9-df29539c6840.png)

That red box is the only thing we can modify for each of these permissions, correct?  If so, perhaps we'd want to slightly extend this resource to allow the flexibility to also define descriptions and decision strategies some time in the future.  Can I get your thoughts on this?

Reply:
Initially my need was just to add a single Policy per Scope. I started from what existed for the `keycloak_identity_provider_token_exchange_scope_permission` resource and with my level of understanding of Go, and Terraform providers, I proposed the client_permission PR and now this one.

Now with your remarks I understand much better what I code and how to do it better :) Thanks 

I have to look at how to use the resource `keycloak_openid_client_authorization_permission` which already exists and is exactly what I should have used from the start :)

Reply:
Sure thing, happy to help!

I will say that I didn't have much involvement with the `keycloak_openid_client_authorization_permission` resource either, so don't be too scared to propose a new / better way of doing things just because that resource doesn't do it.

I think what you have so far is pretty good, but it could be extended a bit further by using sub schemas to allow users to define some of those other properties for a particular policy.  Perhaps something like this:

```hcl
resource "keycloak_users_permissions" {
  realm_id = "master"
  
  view_scope {
    policy_id   = "some policy id" // this is the one attribute you currently support
    description = "foo"            // this is another attribute that could potentially be supported in the future
  }
}
```

So instead of `view_scope_policy_id`, we can use an attribute like `view_scope`, which is a `schema.TypeSet`, to define the policy ID and other attributes.  Or maybe instead of `policy_id = "a string"`, we could do `policies = ["multiple", "policies"]` so that more than one could be supported.

These are just suggestions - perhaps there are better ways to define these attributes.  What do you think about this?